### PR TITLE
Update sbt-mima-plugin to 0.5.0

### PR DESCRIPTION
**Pull request by banno-autobot**
Updates com.typesafe:sbt-mima-plugin from 0.3.0 to 0.5.0.

I'll automatically update this PR to resolve conflicts as long as you don't change it yourself.

If you'd like to skip this version, you can just close this PR. If you have any feedback, just mention @banno-autobot in the comments below.

Have a fantastic day writing Scala!

<details>
<summary>Ignore future updates</summary>

Add this to your `.scala-steward.conf` file to ignore future updates of this dependency:
```
updates.ignore = [{ groupId = "com.typesafe", artifactId = "sbt-mima-plugin" }]
```
</details>

labels: semver-minor

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -172,6 +172,7 @@ lazy val mimaSettings = {
 
   Seq(
     mimaFailOnProblem := mimaVersions(version.value).toList.headOption.isDefined,
+    mimaFailOnNoPrevious in ThisBuild := false,
```
Comment (ChristopherDavenport) [inline build.sbt]: Can we remove the `ThisBuild` and instead disable the plugin on the root project?

File: build.sbt [hunk 2]
```diff
@@ -5,7 +5,7 @@ lazy val `vault4s` = project.in(file("."))
   .aggregate(core, docs)
 
 lazy val core = project.in(file("core"))
-  .settings(commonSettings, releaseSettings, mimaSettings)
```
Comment (ChristopherDavenport) [inline build.sbt]: Why would you remove the mima check from the project?

Comment (ChristopherDavenport) [reply build.sbt]: If you add `disablePlugin(MimaPlugin) to the `vault4s` project that should make it all work.

Comment (JesusMtnez) [reply build.sbt]: :man_facepalming: Thanks!

**Pull request closed without merging**